src/pri-dual-vec.md

```diff
@@ -0,0 +1,193 @@
+# Primal and Dual Vectors
+
+The finite element method uses vectors of data in a variety of ways and the
+differences can be subtle.  MFEM defines `GridFunction`, `LinearForm`, and
+`Vector` classes which help to distinguish the different roles that vectors of
+data can play.
+
+<center>**Graphical summary of Primal, Dual, DoF, True DoF vectors**</center>
+![](img/FEM-LA-Diagrams.png)
+
+## Primal Vectors
+
+The finite element method is based on the notion that a smooth function can be
+approximated by a sum of piece-wise functions called *basis functions*:
+$$f(\vec{x})\approx\sum_i f_i \phi_i(\vec{x}) \label{expan}$$
+The support of an individual basis function, $\;\phi_i(\vec{x})$, will either be
+a single zone or a collection of zones that share a common vertex, edge, or
+face.  The expansion coefficients, $\;f_i$, are linear functionals of the field
+being approximated, $\;f(\vec{x})$ in this case.  The $\;f_i$ could be as simple
+as values of the function at particular points, called interpolation points,
```

Comment:
"and the basis is called a *Nodal basis*, and all the other kind of basis are called *Modal basis*." ("Modal" meaning "Non-nodal")
"Modal basis released from the Nodal constraint are often chosen with problem specific properties, *e.g.* orthogonality of basis function that result in a diagonal mass matrix, similar choices can be made for a sparse Laplacian operator, or even the well-known Bernstein basis possessing the properties of Bernstein polynomials."

Reply:
I will be interested to participate to this. I also had to deal with common pitfalls when talking about function basis/convergence results with physicists.
A common error was that physicists were interpolating with their homebrew interpolation between values given at nodal points, resulting in a big loss of accuracy... ignoring that FE methods converge in L2 norm, not L_inf, meaning that you don't really have point-wise convergence results; or another way to put it is that FE methods give a solution with the same accuracy at any point, nodal points function values are just convenient/inexpensive to compute.